Archive surf update (#1231)
rug-cit-hpc/league-of-robots: roles/online_docs/templates/mkdocs/docs/archive.md

Archive is automatically mounted when user navigates to the `/groups/[GROUP]/arc[XX]` folder. At that moment storage from remote server gets mounted on the folder. It remains accessible until some specific idle time is reached.

Data-manager account of the specific group is the **only** account that has **read** and **write** access to the archive folder of the group. This is to prevent users accidentally recalling files online when not needed. Also to make sure that all the files are stored in correct format (see 'Best practices' below).

## 2. Managing data

> After some time, all the files on remote archive server get automatically migrated to the tape. When this happens, entire tree with all the (sub)folders and files can be normally accessed and browsed. `ls` and `cd` commands work just as they work on *regular* filesystem. File names, permissions and metadata (age, size, ownerhip) can be listed.

Reviewer: ```suggestion
After some time, all the files on remote archive server get automatically migrated to the tape. **metadata** and **data structure** remain remote disks, while **data** content exist only on tape. When this happens, the meta-data (like file names, permissions, timestamps, size, ownerhip, etc.) can be normally accessed and the structure can be normally browsed. `ls` and `cd` commands work just like on *regular* filesystem.
```

> After some time, all the files on remote archive server get automatically migrated to the tape. **metadata** and **data structure** remain remote disks, while **data** content exist only on tape. When this happens, the meta-data (like file names, permissions, timestamps, size, ownerhip, etc.) can be normally accessed and the structure can be normally browsed. `ls` and `cd` commands work just like on *regular* filesystem.

Reviewer: ```suggestion
After some time, all the files on remote archive server get automatically migrated to the tape. The **metadata** and **tree structure** remain on remote disks, while the **data content** exist only on tape. When this happens, the metadata (like file names, permissions, timestamps, size, ownership, etc.) can be normally accessed and the structure can be normally browsed. `cd`, `ls` and `find` commands work just like they do on a *regular* filesystem.
```

> The difference is that the file content is not directly available anymore - until it is recalled. If anything is done on the file content (attempt to compress it or modify with vim/nano) or simply read it (with `less`, `cat` or `grep`), will result in the command line **appear stuck**. Actually it is waiting for data to be retrieved. This takes a long time, since the data is copied from tapes to disks.

Reviewer: ```suggestion
The difference is that the file content is not directly available anymore - until it is recalled. Any command that attempts to read the file content (like `less`, `cat` or `grep`) or edit the content (like `vim` or `nano`) may **appear stuck**. Actually it is waiting for data to be retrieved, which can take a very long time since the data needs to be copied back from tape to disk.
```

> The difference is that the file content is not directly available anymore - until it is recalled. Any command that attempts to read the file content (like `less`, `cat` or `grep`) or edit the content (like `vim` or `nano`) may **appear stuck**. Actually it is waiting for data to be retrieved, which can take a very long time since the data needs to be copied back from tape to disk.

Reviewer: ```suggestion
The difference is that the file content is not directly available anymore - until it is recalled. Any command that attempts to read the file content (like `less`, `cat` or `grep`) or edit the content (like `vim` or `nano`) may **appear stuck**. Actually it is waiting for data to be retrieved, which can take a very long time since the data needs to be copied back from tape to disk.
```
```suggestion
The difference is that the file content is not directly available anymore until it is recalled. Any command that attempts to read the file content (like `less`, `cat` or `grep`) or to edit the content (like `vim` or `nano`) may **appear stuck**. Actually the machine is waiting for data to be retrieved, which can take a very long time since the data needs to be copied back from tape to disk.
```

(optional, but highly recommended) Preperate the data by merging multiple files/folders into one compressed **tar** file
```
   dm-user $ tar -czvf /groups/[group]/[tmp0X]/projects/project-x.tar.gz /groups/[group]/[prm0X]/projects/x/*
```

### Uploading

Upload file(s) to the archive
```

>    dm-user $ cp /groups/[group]/[prm0X]/project-x.tar /groups/[group]/[arc0X]/projects/project-x.tar

Reviewer: Would it make sense to use the previously created
`/groups/[group]/[tmp0X]/projects/project-x.tar.gz`
from the example above?

## 7. Issues

So far most of the bugs have been resolved, but it could happen that

>  - archive folder is not available - please inform helpdesk, this should not happen, but it can be that remote system is temporarily down,

Reviewer: ```suggestion
 - the archive folder is not available - please inform the helpdesk unless maintenance was announced,
```

 - download/upload perfomance occasionally drops - this most probably depends on the Login node usage (and data copy by other users) - notify helpdesk if it persists for a longer period,

>  - submitting the commands did not provide the results - has happened in first implemenentation of the archive solution, but should be fixed now.

Reviewer: As this is fixed now and only applies to a previous implementation: is this relevant to mention?
